### PR TITLE
Sprint12 05 a47457 app tomcat mod jk fix

### PR DESCRIPTION
**Pull request by Neve**
this fixes https://wush.net/trac/rightscale/ticket/4686
the reason of this issue was because tomcat lost connection to apache mod_jk
and work separately from apache.

### CODE REVIEW

File: cookbooks/app_tomcat/README.rdoc [hunk 1]
```diff
@@ -37,8 +37,14 @@ There are no known limitations.
 
 == DETAILS:
 === General
-The <tt>app_tomcat</tt> cookbook will install and setup tomcat application
-server, which can connect to MySQL or Postgres database servers.
+
+The <tt>app_tomcat</tt> cookbook will install and setup Apache web server with mod_jk module
+and Tomcat application server, with support of MySQL or Postgres database servers.
+Cookbook will create separate vhost config for Apache web server, which will allow
```
Comment (douglaswth) [inline cookbooks/app_tomcat/README.rdoc]: Some minor changes: "the Apache web server", "the Tomcat application server", "The Cookbook".

File: cookbooks/app_tomcat/providers/default.rb [hunk 2]
```diff
@@ -161,6 +161,8 @@
     )
   end
 
+  # Define internal port for tomcat it must be different then apache ports
```
Comment (douglaswth) [inline cookbooks/app_tomcat/providers/default.rb]: "Then" should be "than".

File: cookbooks/app_tomcat/providers/default.rb [hunk 3]
```diff
@@ -161,6 +161,8 @@
...
+  tomcat_port = port+1
```
Comment (douglaswth) [inline cookbooks/app_tomcat/providers/default.rb]: Please put spaces between variables, operators, and numbers.

File: cookbooks/app_tomcat/providers/default.rb [hunk 4]
```diff
@@ -269,44 +271,53 @@
...
-      apache_docroot = "#{app_root}"
-    else
-      log "  root_war defined, setting apache docroot to #{app_root}/ROOT"
-      apache_docroot = "#{app_root}/ROOT"
-    end
+  # Apache fix on RHEL
+  file "/etc/httpd/conf.d/README" do
+    action :delete
+    only_if do node[:platform] == "redhat" end
+  end
```
Comment (douglaswth) [inline cookbooks/app_tomcat/providers/default.rb]: What is the reasoning for this removal of a README file? Wouldn't it make more sense to just filter it out of anything that is accidentally reading it as a config file?

Comment (Neve) [inline cookbooks/app_tomcat/providers/default.rb]: in /etc/httpd/conf.d/ we have

```
 ls /etc/httpd/conf.d/
charset
mod_jk.conf
security
status.conf
```

and `Include /etc/httpd/conf.d/` in /etc/httpd/conf/httpd.conf

we will have to rename all configs that do not have ".conf" 
charset ->charset.conf
security -> security.conf

and change `Include /etc/httpd/conf.d/` -> `Include /etc/httpd/conf.d/*.conf`

this changes will affect all templates that use web_apache
maybe it is better to fill ticket and create story for the next sprint?

Comment (Neve) [inline cookbooks/app_tomcat/providers/default.rb]: hmm maybe it just an deadcode. will recheck
nope it is still there

Comment (Neve) [inline cookbooks/app_tomcat/providers/default.rb]: k that is fixed and tested
https://github.com/rightscale/cookbooks/pull/2

File: cookbooks/app_tomcat/README.rdoc [hunk 5]
```diff
@@ -37,8 +37,14 @@ There are no known limitations.
 
 == DETAILS:
 === General
-The <tt>app_tomcat</tt> cookbook will install and setup tomcat application
-server, which can connect to MySQL or Postgres database servers.
+
+The <tt>app_tomcat</tt> cookbook will install and setup the Apache web server with mod_jk module
+and the Tomcat application server, with support of MySQL or Postgres database servers.
```
Comment (lopaka) [inline cookbooks/app_tomcat/README.rdoc]: '...with support for MySQL..."

**Pull request merged**